src/pages/workspace/WorkspaceMembersPage.js

```diff
@@ -284,226 +250,158 @@ function WorkspaceMembersPage(props) {
-            return;
-        }
-        removableMembers[member.accountID] = member;
-    });
     const policyID = lodashGet(props.route, 'params.policyID');
     const policyName = lodashGet(props.policy, 'name');
 
+    const getMemberOptions = () => {
+        let result = [];
+
+        _.each(props.policyMembers, (policyMember, accountID) => {
+            if (isDeletedPolicyMember(policyMember)) {
+                return;
+            }
+
+            const details = props.personalDetails[accountID];
+
+            if (!details) {
+                Log.hmmm(`[WorkspaceMembersPage] no personal details found for policy member with accountID: ${accountID}`);
+                return;
+            }
+
+            // If search value is provided, filter out members that don't match the search value
+            if (searchValue.trim()) {
+                let memberDetails = '';
+                if (details.login) {
+                    memberDetails += ` ${details.login.toLowerCase()}`;
+                }
+                if (details.firstName) {
+                    memberDetails += ` ${details.firstName.toLowerCase()}`;
+                }
+                if (details.lastName) {
+                    memberDetails += ` ${details.lastName.toLowerCase()}`;
+                }
+                if (details.displayName) {
+                    memberDetails += ` ${details.displayName.toLowerCase()}`;
+                }
+                if (details.phoneNumber) {
+                    memberDetails += ` ${details.phoneNumber.toLowerCase()}`;
+                }
+
+                if (!OptionsListUtils.isSearchStringMatch(searchValue.trim(), memberDetails)) {
+                    return;
+                }
+            }
+
+            // If this policy is owned by Expensify then show all support (expensify.com or team.expensify.com) emails
+            // We don't want to show guides as policy members unless the user is a guide. Some customers get confused when they
+            // see random people added to their policy, but guides having access to the policies help set them up.
+            if (PolicyUtils.isExpensifyTeam(details.login || details.displayName)) {
+                if (policyOwner && currentUserLogin && !PolicyUtils.isExpensifyTeam(policyOwner) && !PolicyUtils.isExpensifyTeam(currentUserLogin)) {
+                    return;
+                }
+            }
+
+            const isAdmin = props.session.email === details.login || policyMember.role === CONST.POLICY.ROLE.ADMIN;
+
+            result.push({
+                keyForList: accountID,
+                isSelected: _.contains(selectedEmployees, Number(accountID)),
+                isDisabled: accountID === props.session.accountID || details.login === props.policy.owner || policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
```

Comment:
I think the `isDisabled` check contributed to creating the following issue:
- https://github.com/Expensify/App/issues/38925

but since the transition from optimistic local data to real BE returned data wasn't tested when this PR was merged -> it wasn't reported as an issue until recently.